lib/at_host/at_host.c

```diff
@@ -53,28 +53,60 @@ enum select_uart {
 
 static enum term_modes term_mode;
 static struct device *uart_dev;
-static int at_socket_fd = INVALID_DESCRIPTOR;
-static struct pollfd fds[1];
-static int nfds;
 static u8_t at_buf[AT_MAX_CMD_LEN];
 static size_t at_buf_len;
-static struct k_work at_cmd_send_work;
-static struct k_thread socket_thread;
-static K_THREAD_STACK_DEFINE(socket_thread_stack, THREAD_STACK_SIZE);
+static struct k_work cmd_send_work;
+static const char termination[3] = { '\0', '\r', '\n' };
 
+static inline void write_uart_string(char *str, size_t len)
+{
+	for (size_t i = 0; i < len; i++) {
+		uart_poll_out(uart_dev, str[i]);
+	}
+}
 
-static const char termination[3] = { '\0', '\r', '\n' };
+static void response_handler(char *response)
+{
+	int len = strlen(response) + 1;
+	/* Forward the data over UART */
+	if (len > 0) {
+		write_uart_string(response, len);
+	}
+}
 
-static void at_cmd_send(struct k_work *work)
+static void cmd_send(struct k_work *work)
 {
-	int bytes_sent;
+	size_t            chars;
+	char              str[15];
+	enum at_cmd_state state;
+	int               err;
 
 	ARG_UNUSED(work);
 
-	bytes_sent = send(at_socket_fd, at_buf, at_buf_len, 0);
+	err = at_cmd_write_with_callback(at_buf, response_handler, &state);
+	if (state != AT_CMD_OK) {
```

Comment:
Why not just `if (err)`?

Reply:
No good reason that I can come up with, I think I just wanted to align with with the switch-case. Anyway changed to use err now.